src/wspace_client_scout.cc

```diff
@@ -153,31 +194,32 @@ void* WspaceClient::RxRcvAth(void* arg) {
     }
 #endif
 
-    if (laptop == kFrontLaptop)
-      raw_pkt_front_buf_.PushPkts(hdr->raw_seq(), true/**is good*/);
-    else if (laptop == kBackLaptop)
-      raw_pkt_back_buf_.PushPkts(hdr->raw_seq(), true);  
+    hdr->ParseHeader(&batch_id_parse, &start_seq_parse, &coding_index_parse, &k, &n, &bs_id);
+    //if (type == Tun::kWspace) {
+      radio_context_tbl_[*radio_id]->raw_pkt_buf()->PushPkts(hdr->raw_seq(), true/**is good*/, bs_id);
+    //}
     /** else, do nothing for the cellular case.*/
 
-    hdr->ParseHeader(&batch_id_parse, &start_seq_parse, &coding_index_parse, &k, &n);
 #ifdef WRT_DEBUG
-    printf("Receive: laptop: %d raw_seq: %u batch_id: %u seq_num: %u start_seq: %u coding_index: %d k: %d n: %d\n", 
-    laptop, hdr->raw_seq(), batch_id_parse, start_seq_parse + coding_index_parse, start_seq_parse, 
+    printf("Receive from bs_id:%d via radio_id: %d raw_seq: %u batch_id: %u seq_num: %u start_seq: %u coding_index: %d k: %d n: %d\n", 
+      bs_id, *radio_id, hdr->raw_seq(), batch_id_parse, start_seq_parse + coding_index_parse, start_seq_parse, 
     coding_index_parse, k, n);
 #endif
     uint32 per_pkt_duration = (nread * 8.0) / (hdr->GetRate() / 10.0) + DIFS_80211ag + SLOT_TIME * 5;  /** in us.*/
 
     if (batch_id > batch_id_parse) {  /** Out of batch order.*/
-      if (laptop == kCellular) {
+      /*
+      if (type == Tun::kCellular) {
         assert(coding_index_parse < k);
         seq_num = start_seq_parse + coding_index_parse;
         uint16 len = hdr->lens()[coding_index_parse];
-        data_pkt_buf_.EnqueuePkt(seq_num, len, (char*)hdr->GetPayloadStart());
+        radio_context_tbl_[*radio_id]->data_pkt_buf()->EnqueuePkt(seq_num, len, (char*)hdr->GetPayloadStart());
```

Comment:
Check client_id before enqueue. Make sure to deal with multiple Enqueues in different conditions.